v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -57,7 +62,196 @@ async function getTasksAndHreEnvironment(
+        const command =
+          "npx hardhat --config ./user-hardhat.config.ts user-task";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults(
+          "user-hardhat.config.ts",
+        );
+        assert.equal(results[0], true);
+      });
+    });
+
+    describe("one of the hardhat default task with global flags and parameters", async function () {
+      useFixtureProject("cli/parsing/base-project");
+
+      // TODO: update with a real task as soon as they are implemented
+      it("should run one of the hardhat default task", async function () {
+        const m = mock.method(console, "log", () => {});
+
+        const command = "npx hardhat --show-stack-traces example";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        assert.equal(m.mock.calls.length, 4);
+        assert.equal(m.mock.calls[2].arguments[0], "from a plugin");
+      });
+    });
+
+    describe("task with global flags and parameters", async function () {
+      useFixtureProject("cli/parsing/tasks-and-subtasks");
+
+      it("should run the task with global flags and parameters", async function () {
+        const command =
+          "npx hardhat --show-stack-traces task --param1 <value1> <value2> <value3>";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults();
+        assert.deepEqual(results, [true, true, true]);
+      });
+
+      it("should run the task with the default value", async function () {
+        const command = "npx hardhat task-default --show-stack-traces";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults();
+        assert.deepEqual(results, [true, false, false]);
+      });
+
+      it("should run the subtask with global flags and parameters", async function () {
+        const command =
+          "npx hardhat task subtask --param1 <value1> --show-stack-traces <value2> <value3>";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults();
+        assert.deepEqual(results, [true, true, true]);
+      });
+
+      it("should run the subtask with the default value", async function () {
+        const command =
+          "npx hardhat task-default --show-stack-traces subtask-default";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults();
+        assert.deepEqual(results, [true, false, false]);
+      });
+    });
+
+    // TODO: as soon as the 'help task' is done, this test should be updated
```

Comment:
@zoeyTM I know that you're currently working on the help task. I created this test and the following ones as placeholder so you just need to modify them later (or copy the logic)